include/bluetooth/audio/has.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_HAS_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_HAS_H_
+
+/**
+ * @brief Hearing Access Service (HAS)
+ *
+ * @defgroup bt_has Hearing Access Service (HAS)
+ *
+ * @ingroup bluetooth
+ * @{
+ *
+ * The Hearing Access Service is used to identify a hearing aid and optionally
+ * to control hearing aid presets.
+ *
+ * [Experimental] Users should note that the APIs can change as a part of
+ * ongoing development.
+ */
+
+#include <bluetooth/conn.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Opaque Hearing Access Service object. */
+struct bt_has;
+
+/** Hearing Aid device type */
+enum bt_has_hearing_aid_type {
+	BT_HAS_HEARING_AID_TYPE_BINAURAL,
+	BT_HAS_HEARING_AID_TYPE_MONAURAL,
+	BT_HAS_HEARING_AID_TYPE_BANDED,
+};
+
+/** Hearing Aid device capablilities */
+enum bt_has_capabilities {
+	BT_HAS_PRESET_SUPPORT = BIT(0),
+};
+
+/** @brief Hearing Access Service Client callback structure. */
+struct bt_has_client_cb {
+	/**
+	 * @brief Callback function for bt_has_discover.
+	 *
+	 * This callback is called when discovery procedure is complete.
+	 *
+	 * @param conn Bluetooth connection object.
+	 * @param has Pointer to the Hearing Access Service object.
+	 * @param type Hearing Aid type.
+	 * @param caps Hearing Aid capabilities.
+	 */
+	void (*discover)(struct bt_conn *conn, struct bt_has *has,
+			 enum bt_has_hearing_aid_type type, enum bt_has_capabilities caps);
```

Comment:
Error case is not documented. `has` would be NULL in case of error, right? 

Would it make sense to report the ATT/ERRNO error directly? If it's an issue with e.g. pairing/bonding, then an application can possibly handle that.

Reply:
> Can do that. Regarding pairing/bonding issue I think Zephyr can handle that internally - once insufficient encryption/authentication error is returned by server, the pairing (or encryption) is started autonomously.

Correct, but that may fail if e.g. our bonding storage is full or due to other issues during pairing, in which case an application may want to reconnect or clear up bonding storage manually.